streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -296,7 +296,8 @@ public StreamThread(TopologyBuilder builder,
         this.processId = processId;
         this.partitionGrouper = config.getConfiguredInstance(StreamsConfig.PARTITION_GROUPER_CLASS_CONFIG, PartitionGrouper.class);
         this.streamsMetadataState = streamsMetadataState;
-        threadClientId = clientId + "-" + threadName;
+        //clientId may already contain application id, removing to avoid duplication
+        threadClientId = clientId.replace(applicationId, "") + "-" + threadName;
```

Comment:
Good point. We should keep app.id for the log messages.
So only add `final` in line 291 and apply `threadName.replace(applicationId + "-", "")`  in line 300.